### PR TITLE
Make extended attribute writing more robust against filesystem limits.

### PR DESCRIPTION
**Pull request by eric321**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Some filesystems limit the size of extended attributes (e.g., 4kB on ext4). The description can easily get close to or exceed this. By writing it last, we avoid missing out on the other ones.

Also catch each write_xattr() failure individually, so we can report which attributes were not written.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Review by Grub4K — CHANGES_REQUESTED** [review]
No reason to change it to a list, dicts are ordered

**Comment by eric321** [comment]
done

**Comment by Grub4K** [comment]
Try to avoid force pushing, prs will be squashed on merge, so take as many commits as you want

File: yt_dlp/postprocessor/xattrpp.py [hunk 1]
```diff
@@ -29,35 +29,38 @@ class XAttrMetadataPP(PostProcessor):
         # 'user.xdg.comment': 'description',
```
Comment (bashonly) [inline yt_dlp/postprocessor/xattrpp.py:29]: I know this is old/dead code but let's move this down to the end of the dict
```suggestion
```

File: yt_dlp/postprocessor/xattrpp.py [hunk 2]
```diff
@@ -29,35 +29,38 @@ class XAttrMetadataPP(PostProcessor):
...
         'user.dublincore.title': 'title',
         'user.dublincore.date': 'upload_date',
-        'user.dublincore.description': 'description',
         'user.dublincore.contributor': 'uploader',
         'user.dublincore.format': 'format',
+        # We do this last because it may get us close to the xattr limits
+        # (e.g., 4kB on ext4), and we don't want to have the other ones fail
+        # because of that.
+        'user.dublincore.description': 'description',
```
Comment (bashonly) [inline yt_dlp/postprocessor/xattrpp.py:37]:
```suggestion
        'user.dublincore.description': 'description',
        # 'user.xdg.comment': 'description',
```

File: yt_dlp/postprocessor/xattrpp.py [hunk 3]
```diff
@@ -29,35 +29,38 @@ class XAttrMetadataPP(PostProcessor):
...
-                tip = ('You need to use NTFS' if compat_os_name == 'nt'
-                       else 'You may have to enable them in your "/etc/fstab"')
-                raise PostProcessingError(f'This filesystem doesn\'t support extended attributes. {tip}')
+            except XAttrUnavailableError as e:
+                raise PostProcessingError(str(e))
+            except XAttrMetadataError as e:
+                if e.reason == 'NO_SPACE':
+                    self.report_warning(
+                        'There\'s no disk space left, disk quota exceeded or filesystem xattr limit exceeded. '
+                        f'Extended attribute {xattrname} was not written.')
```
Comment (bashonly) [inline yt_dlp/postprocessor/xattrpp.py:57]:
```suggestion
                        f'Extended attribute "{xattrname}" was not written.')
```

File: yt_dlp/postprocessor/xattrpp.py [hunk 4]
```diff
@@ -29,35 +29,38 @@ class XAttrMetadataPP(PostProcessor):
...
+                elif e.reason == 'VALUE_TOO_LONG':
+                    self.report_warning(f'Unable to write extended attribute {xattrname} due to too long values.')
```
Comment (bashonly) [inline yt_dlp/postprocessor/xattrpp.py:59]:
```suggestion
                    self.report_warning(f'Unable to write extended attribute "{xattrname}" due to too long values.')
```

**Review by bashonly — APPROVED** [review]

**Review by Grub4K — APPROVED** [review]

**Pull request merged**